src/Servers/Kestrel/tools/CodeGenerator/HttpProtocolFeatureCollection.cs

```diff
@@ -74,7 +74,13 @@ public static string GenerateFile()
                 "IRouteValuesFeature",
                 "IEndpointFeature"
             };
-            
+
+            // NOTE: Each item in this list MUST always be reset by each protocol in their OnReset() method
+            var skipResetFeatures = new[]
```

Comment:
This is my only concern with this change, how do we not forget about these? Maybe something to follow up on https://github.com/dotnet/aspnetcore/issues/31323

Reply:
Same issue a couple lines up that would probably want to solve at the same time?
https://github.com/dotnet/aspnetcore/blob/7dea0cb6736bc8ea2b53e5a716b926e7a80a4430/src/Servers/Kestrel/tools/CodeGenerator/HttpProtocolFeatureCollection.cs#L57-L58

Reply:
Though at least you get a compile error now if you include features which aren't implemented https://github.com/dotnet/aspnetcore/pull/31322#pullrequestreview-622710999

Reply:
Ah, this is like the other null features which are then implemented downlevel; will remove the skip

Reply:
Done